usr/lib/zypp/plugins/urlresolver/susecloud

```diff
@@ -31,34 +31,36 @@ utils.set_proxy()
 class SUSECloudPlugin(Plugin):
     def RESOLVEURL(self, headers, body):
         """Convert the URL for the repo"""
-        update_server = utils.get_smt()
-        server_name = ''
-        if not update_server:
-            # Something went seriously wrong, however try the name from
-            # the hosts file
-            server_name = utils.get_update_server_name_from_hosts()
-        else:
-            server_name = update_server.get_FQDN()
-        if server_name:
-            srv_url = 'https://%s' % server_name
-            repo_url = srv_url + headers.get('path')
-            credentials = headers.get('credentials')
-            if credentials:
-                repo_url += '?credentials=' + credentials
-            instance_data = b''
-            try:
-                instance_data = bytes(
-                    utils.get_instance_data(utils.get_config()),
-                    'utf-8'
-                )
-            except TypeError:
-                logging.warning(
-                    '[URL-Resolver] Unable to retrieve instance data'
-                )
-            # zypper wants a string or bad things happen on the other side
-            inst_data_bin_str = base64.b64encode(instance_data).decode('utf-8')
-        else:
-            repo_url = None
+        repo_url = None
+        if not utils.is_new_registration():
+            update_server = utils.get_smt()
+            server_name = ''
+            if not update_server:
+                # Something went seriously wrong, however try the name from
+                # the hosts file
+                server_name = utils.get_update_server_name_from_hosts()
+            else:
+                server_name = update_server.get_FQDN()
+            if server_name:
```

Comment:
Can dedent this if block? and move `server_name = ''` below `repo_url = None`?

Reply:
Yes `server_name` could be initialized outside the `if not utils.is_new_registration()` condition. Then the check for `if server_name` could move outside the `if not utils.is_new_registration()` check.

However conceptually we want to resolve the URL only when it is not a "first_time" invocation of the resolver. Thus conceptually that would hen imply 

```
if utils.is_new_registration():
    self.answer(
            'RESOLVEDURL',
            {'X-Instance-Data': inst_data_bin_str},
            repo_url
        )
else:
    EVERYTHING_ELSE
```

We can of course argue about which is easier to read, but that just a mounts to a style choice with no "right" or "wrong" way.

Anyway, thanks for raising the question it made be realize that `inst_data_bin_str` is not properly initialized.